Sources/SendGridKit/SendGridClient.swift

```diff
@@ -21,31 +21,34 @@ public struct SendGridClient: Sendable {
         return decoder
     }()
 
-    public init(httpClient: HTTPClient, apiKey: String) {
+    /// Initialize a new `SendGridClient`
+    ///
+    /// - Parameters:
+    ///   - httpClient: The `HTTPClient` to use for sending requests
+    ///   - apiKey: The SendGrid API key
+    ///   - forEU: Whether to use the API endpoint for global users and subusers or for EU regional subusers
+    public init(httpClient: HTTPClient, apiKey: String, forEU: Bool = false) {
         self.httpClient = httpClient
         self.apiKey = apiKey
+        self.apiURL = forEU ? "https://api.eu.sendgrid.com/v3/mail/send" : "https://api.sendgrid.com/v3/mail/send"
     }
```

Comment:
Are there only these 2 endpoints / regions? If not, I'd allow passing apiURL directly.
Perhaps with static funcs like `static func withGlobalEndpoint` or something (not a fan of using `with`-style functions since they are becoming a convention, but not sure what else to use)

Reply:
Optimally the init accepts a `region` and figures out the endpoint by that.
It's likely sendGrid has documented an exhaustive list of the regions they support.

Reply:
Yeah only a global and a European one

Reply:
https://www.twilio.com/docs/sendgrid/api-reference/mail-send/mail-send